uaclient/security_status.py

```diff
@@ -154,6 +157,7 @@ def filter_updates(
         for package in packages:
             # We only care about installed packages here
             if package.current_ver:
+                candidate_version = policy.get_candidate_ver(package)
```

Comment:
Sorry I know this is still a WIP.

I'd say this function is meant to filter the available updates for a package list, and the candidate logic should not live there.
I'd say the `create_updates_list` function is the one which should have knowledge about the candidates, so it can be passed to `get_update_status` and treated there as a separate case, instead of a separate variable drilled down.
I know depcaches have candidates for packages - I suggest a small helper function:
`_is_version_the_candidate(version: apt_pkg.Version) -> bool:`
which would check the parent package candidate and see it it matches the version, or something close to this.
Also, consider `lru_cache`-ing the logic which finds a candidate for a package, so you don't have to invoke the depcache multiple times for the same package (although this should be minimal / fine for this step)

Reply:
Good cache (catch) :eyes: 
Yeah since we discussed not going the policy route, it makes sense to get a different function and not populate the filter function with this task.